### PR TITLE
connect/ca: require new vault mount points when updating the key type/bits for the vault connect CA provider

### PR DESCRIPTION
**Pull request by rboyer**
progress on #9572

### CODE REVIEW

**Comment by rboyer** [comment]
This is similar in spirit to #10330

File: agent/consul/connect_ca_endpoint_test.go [hunk 1]
```diff
@@ -512,6 +514,114 @@ func TestConnectCAConfig_TriggerRotation(t *testing.T) {
...
+	defer os.RemoveAll(dir1)
+	defer s1.Shutdown()
+
+	codec := rpcClient(t, s1)
+	defer codec.Close()
+
+	testrpc.WaitForTestAgent(t, s1.RPC, "dc1")
+
+	// Capture the current root.
+	// var originalRoot *structs.CARoot
```
Comment (dnephin) [inline agent/consul/connect_ca_endpoint_test.go:551]: Stray debug comments?

```suggestion
```

File: agent/consul/connect_ca_endpoint_test.go [hunk 2]
```diff
@@ -512,6 +514,114 @@ func TestConnectCAConfig_TriggerRotation(t *testing.T) {
...
+	{
+		rootList, _, err := getTestRoots(s1, "dc1")
+		require.NoError(t, err)
+		require.Len(t, rootList.Roots, 1)
+		// originalRoot = activeRoot
```
Comment (dnephin) [inline agent/consul/connect_ca_endpoint_test.go:556]:
```suggestion
```

File: agent/consul/connect_ca_endpoint_test.go [hunk 3]
```diff
@@ -512,6 +514,114 @@ func TestConnectCAConfig_TriggerRotation(t *testing.T) {
...
+			Provider: "vault",
+			Config: map[string]interface{}{
+				"Address":             testVault.Addr,
+				"Token":               testVault.RootToken,
+				"RootPKIPath":         "pki-root/",
+				"IntermediatePKIPath": "pki-intermediate/",
+			},
+		}
+	})
+	defer os.RemoveAll(dir1)
```
Comment (dnephin) [inline agent/consul/connect_ca_endpoint_test.go:542]: minor: these remove lines are no longer necessary after we started using `t.Cleanup` in the test helper that creates the temp directory.

```suggestion
```

**Review by dnephin — APPROVED** [review]
LGTM! This seems like it fixes the UX problem for the vault provider by refusing the update when it can't be fully applied. The error message should make it clear to the user how to resolve the issue.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/408180.

**Pull request merged**